industrial_extrinsic_cal/src/ros_camera_observer.cpp

```diff
@@ -205,322 +212,349 @@ int ROSCameraObserver::getObservations(CameraObservations &cam_obs)
       break;
     case pattern_options::CircleGrid:
       if (sym_circle_) // symetric circle grid
-	{
-	  ROS_DEBUG_STREAM("Finding Circles in grid, symmetric...");
-	  if(use_circle_detector_){
-	    successful_find = cv::findCirclesGrid(image_roi_, pattern_size, observation_pts_, 
-						  cv::CALIB_CB_SYMMETRIC_GRID,
-						  circle_detector_ptr_);
-	  }
-	  else{
-	    successful_find = cv::findCirclesGrid(image_roi_, pattern_size, observation_pts_, cv::CALIB_CB_SYMMETRIC_GRID);
-	  }
-	}
+      {
+        ROS_DEBUG_STREAM("Finding Circles in grid, symmetric...");
+        if(use_circle_detector_){
+          successful_find = cv::findCirclesGrid(image_roi_, pattern_size, observation_pts_,
+              cv::CALIB_CB_SYMMETRIC_GRID, circle_detector_ptr_);
+          if(!successful_find)
+          {
+            successful_find = cv::findCirclesGrid(image_roi_, pattern_size_flipped, observation_pts_,
+                cv::CALIB_CB_SYMMETRIC_GRID, circle_detector_ptr_);
+            flipped_successful_find = successful_find;
+          }
+        }
+        else{
+          successful_find = cv::findCirclesGrid(image_roi_, pattern_size, observation_pts_, cv::CALIB_CB_SYMMETRIC_GRID);
+          if(!successful_find){
+            successful_find = cv::findCirclesGrid(image_roi_, pattern_size_flipped, observation_pts_, cv::CALIB_CB_SYMMETRIC_GRID);
+            flipped_successful_find = successful_find;
+          }
+        }
+      }
       else         // asymetric circle grid
-	{
-	  ROS_DEBUG_STREAM("Finding Circles in grid, asymmetric...");
-	  if(use_circle_detector_){
-	    successful_find = cv::findCirclesGrid(image_roi_, pattern_size, observation_pts_, 
-						  cv::CALIB_CB_ASYMMETRIC_GRID | cv::CALIB_CB_CLUSTERING,
-						  circle_detector_ptr_);
-	  }
-	  else{
-	    successful_find = cv::findCirclesGrid(image_roi_, pattern_size , observation_pts_, 
-						  cv::CALIB_CB_ASYMMETRIC_GRID | cv::CALIB_CB_CLUSTERING);
-	  }
+      {
```

Comment:
I have not check the asymmetric target finder.  I made the changes so that it could find the target if it happened to be rotated 90 degrees (origin dot is not in the bottom left corner of the image).  Would you prefer that I not make the changes to the asymmetric target finder section?  If no one has tested this case, why is it in the code?